src/main.rs

```diff
@@ -225,25 +227,24 @@ fn set_new_password(db_conn: &str, uname: &str, password_hash: &str) -> Result<(
 fn hash_password(password: &str, pepper: &str, rounds: u32) -> String {
     // This function closely mimics
     // https://github.com/matrix-org/synapse/blob/9bba6ebaa903a81cd94fada114aa71e20b685adb/scripts/hash_password
-    let mut salt = [0u8; 16];
-    let mut rng = OsRng::new().unwrap();
-    rng.fill_bytes(&mut salt[..]);
-    let mut output = [0u8; 24];
-
-    // ... See https://github.com/pyca/bcrypt/blob/fcebaa0db74dc822877128e57a79dcfda2a2dc4f/src/bcrypt/__init__.py#L66-L72
-    // and https://github.com/DaGenix/rust-crypto/blob/cc1a5fde1ce957bd1a8a2e30169443cdb4780111/src/bcrypt.rs#L26
     let peppered_password = format!("{}{}", password, pepper);
-    let bcryptable_password = &peppered_password.as_bytes()[0..72];
-
-    bcrypt::bcrypt(rounds,
-                   &salt[..],
-                   bcryptable_password,
-                   &mut output[..]);
 
-    // rust-crypto doesn't do anything nice for us so we have to format our own hash output
-    let salt_hex = salt.iter().map(|b| format!("{:X}", b)).collect::<Vec<String>>().join("");
-    let out_hex = output.iter().map(|b| format!("{:X}", b)).collect::<Vec<String>>().join("");
-    format!("$2b${}${}$", salt_hex, out_hex)
+    let result = bcrypt::hash(peppered_password.as_ref(), rounds);
+    // unly possible error is cost being bad, already validated. TODO, this can be improved by
```

Comment:
unly?

By "cost being bad" do you mean the amount of rounds? I'm confused by this comment.